apps/nextjs/src/stores/chatStore/actionFunctions/handleExecuteQueuedAction.ts

```diff
@@ -0,0 +1,31 @@
+export function handleExecuteQueuedAction(set, get) {
+  return async () => {
+    const { queuedUserAction, isExecutingAction } = get();
```

Comment:
I think we should destructure the actions like append and reload up here as well